### PR TITLE
MGMT-2297 Cluster monitor should filter our installed and clusters in…

### PR DESCRIPTION
**Pull request by gamli75**
… error

I filtered all cluster states that stay in the same state based on the machine state  - @filanov please review

### CODE REVIEW

File: internal/cluster/cluster.go [hunk 1]
```diff
@@ -265,7 +265,16 @@ func (m *Manager) ClusterMonitoring() {
 	defer func() {
 		m.prevMonitorInvokedAt = curMonitorInvokedAt
 	}()
-	if err = m.db.Find(&clusters).Error; err != nil {
+
+	//no need to refresh cluster status if the cluster is in the following statuses
+	noNeedToMonitorInStates := []string{
+		models.ClusterStatusPreparingForInstallation,
+		models.ClusterStatusFinalizing,
```
Comment (filanov) [inline internal/cluster/cluster.go:272]: Don't we have timeout for both `ClusterStatusPreparingForInstallation` `ClusterStatusFinalizing` ?

File: internal/cluster/cluster_test.go [hunk 2]
```diff
@@ -288,6 +288,31 @@ var _ = Describe("cluster monitor", func() {
...
+				PullSecretSet:      true,
+			}}
+
+			Expect(db.Create(&c).Error).ShouldNot(HaveOccurred())
+			Expect(err).ShouldNot(HaveOccurred())
+			mockMetric.EXPECT().ClusterInstallationFinished(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+		})
+
+		It("installed -> installed", func() {
+			createHost(id, "installed", db)
```
Comment (filanov) [inline internal/cluster/cluster_test.go:309]: please use consts not strings.

File: internal/cluster/cluster.go (hunk 1 above)
Comment (gamli75) [reply internal/cluster/cluster.go:272]: thanks, fixed it

File: internal/cluster/cluster_test.go (hunk 2 above)
Comment (gamli75) [reply internal/cluster/cluster_test.go:309]: fixed

**Review by filanov — APPROVED** [review]

**Pull request merged**